src/IO/WriteBuffer.h

```diff
@@ -150,7 +172,12 @@ class WriteBuffer : public BufferBase
         next();
     }
 
+    virtual void cancelImpl()
```

Comment:
actually why not do the same in read buffers?

Reply:
Read buffers are suppose to be cont methods related to the storage. It is just OK to destroy them at any point.

WriteBuffers should be destroyed only after `finalize` call. It is a source of all difficulties. But that `finalize` call is essential.